src/System.Text.Json/tests/Serialization/TestClasses.cs

```diff
@@ -175,6 +175,162 @@ public void Verify()
         }
     }
 
+    public class SimpleTestClassWithObject : ITestClass
```

Comment:
```C#
public class Person
{
    public string Name { get; set; }
    public Person Spouse { get; set; }
    public Person[] Children { get; set; }
}

public class Employee : Person
{
    public string Title { get; set; }
    public Person Boss { get; set; }
}

Employee carol = new Employee { Name = "Carol", Title = "Realtor" };
Employee mike = new Employee { Name = "Mike", Title = "Architect" };
Person marcia = new Person { Name = "Marcia" };
Employee mikeBoss = new Employee { Name = "Mike's Unnamed Boss" };
Employee alice = new Employee { Name = "Alice", Title = "Alice" };

alice.Boss = carol;
carol.Spouse = mike;
mike.Boss = mikeBoss;
carol.Children = new Person[] { marcia };
```

What does `ToBytes<Person>(alice)` produce?  What about `ToBytes<object>(alice)`?  (Is object polymorphism "deep", or "shallow"?).  Feel free to include the rest of the Brady's, I got bored :smile:

Reply:
Note the above model has a cycle from Person.Children->Person which isn't supported.

Removing the `Children` property, `ToString<Person>(alice)` produces
```{\"Name\":\"Alice\",\"Spouse\":null}```
and `ToString<object>(alice)` produces
```{\"Title\":\"Alice\",\"Boss\":{\"Name\":\"Carol\",\"Spouse\":{\"Name\":\"Mike\",\"Spouse\":null}},\"Name\":\"Alice\",\"Spouse\":null}```

Also note there are additional tests written earlier that verify the static behavior.

Object polymorphism is "shallow" meaning every property that wants polymorphic behavior must be `object` or use the future attribute to enable this on a per-property level.